### PR TITLE
use aws region from controller context

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards Node Pools. This is an effort to drop the `Cluster` CR involvement from the TCNP related stack management.

### CODE REVIEW

File: service/controller/clusterapi/v29/resource/tccp/testdata/case-0-basic-test.golden [hunk 1]
```diff
@@ -4,7 +4,7 @@ Outputs:
   DockerVolumeResourceName:
     Value: rsc-abbacd01
   MasterImageID:
-    Value: ami-0eb0d9bb7ad1bd1e9
+    Value:
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/resource/tccp/testdata/case-0-basic-test.golden]: And look, I fixed something. 🙈

File: service/controller/clusterapi/v29/resource/tcnp/create.go [hunk 2]
```diff
@@ -224,12 +224,12 @@ func newIAMPolicies(ctx context.Context, cl v1alpha1.Cluster, md v1alpha1.Machin
 		Cluster: template.ParamsMainIAMPoliciesCluster{
 			ID: key.ClusterID(&md),
 		},
-		EC2ServiceDomain: key.EC2ServiceDomain(cl),
+		EC2ServiceDomain: key.EC2ServiceDomain(cc.Status.TenantCluster.AWS.Region),
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/resource/tcnp/create.go]: In the next PR we can drop the `Cluster` CR magic.

File: service/controller/clusterapi/v29/encrypter/vault/vault.go [hunk 3]
```diff
@@ -146,22 +146,22 @@ func (e *Encrypter) Encrypt(ctx context.Context, key, plaintext string) (string,
 	return ciphertext, nil
 }
 
-func (e *Encrypter) EnsureCreatedAuthorizedIAMRoles(ctx context.Context, customObject v1alpha1.Cluster) error {
-	err := e.ensureToken()
+func (e *Encrypter) EnsureCreatedAuthorizedIAMRoles(ctx context.Context, cr v1alpha1.Cluster) error {
```
Comment (rossf7) [inline service/controller/clusterapi/v29/encrypter/vault/vault.go]: <3

File: service/controller/clusterapi/v29/resource/tccp/testdata/case-0-basic-test.golden (hunk 1 above)
Comment (xh3b4sd) [reply service/controller/clusterapi/v29/resource/tccp/testdata/case-0-basic-test.golden]: Ah, the heat. After my run I came back to realize this did not fix but break something. Fixing.

**Pull request merged**